lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -124,23 +130,39 @@ export default function getRecipeInterface(
             sessionData?: any;
             userContext: any;
         }): Promise<Session> {
-            const disableAntiCSRF = config.getTokenTransferMethod({ req, userContext }) === "header";
+            logDebugMessage("createNewSession: Started");
+            let outputTransferMethod = config.getTokenTransferMethod({ req, userContext });
+            logDebugMessage("createNewSession: got transfer method " + outputTransferMethod);
+            if (outputTransferMethod === "missing_auth_header") {
+                outputTransferMethod = "header";
+            }
+            logDebugMessage("createNewSession: using transfer method " + outputTransferMethod);
+
+            const disableAntiCSRF = outputTransferMethod === "header";
             let response = await SessionFunctions.createNewSession(
                 helpers,
                 userId,
                 disableAntiCSRF,
                 accessTokenPayload,
                 sessionData
             );
-            attachCreateOrRefreshSessionResponseToExpressRes(config, req, res, response, userContext);
+
+            for (const transferMethod of availableTokenTransferMethods) {
+                if (transferMethod !== outputTransferMethod) {
+                    clearSession(config, req, res, userContext, transferMethod);
```

Comment:
What if the existing session being cleared is the legacy session?

Reply:
1. In most cases a refresh call is made before login, which clears/migrates the legacy session:
- doesSessionExist & getAccessTokenPayload securely is called by `SessionAuth`
- even in custom UIs, they are likely to be called (check the logged-in state and or/get a user id) before showing a sign-in screen
- if any call is made to an endpoint that uses `verifySession` it'll trigger a refresh.
2. Even if it's not, the first call to a verified endpoint will trigger a refresh that cleans the cookie of the old session
3. The old cookie is actually safe to ignore